src/app/pages/quickMaps/pages/baselineDetails/foodItems/foodItems.component.ts

```diff
@@ -1,53 +1,92 @@
+import { CurrentDataService } from 'src/app/services/currentData.service';
+import { QuickMapsService } from '../../../quickMaps.service';
+import 'chartjs-chart-treemap';
+import { ChartData, ChartDataSets, ChartPoint, ChartTooltipItem } from 'chart.js';
 @Component({
   selector: 'app-food-items',
   templateUrl: './foodItems.component.html',
   styleUrls: ['./foodItems.component.scss'],
 })
-export class FoodItemsComponent implements OnInit, AfterViewInit {
+export class FoodItemsComponent implements OnInit {
   @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
-  public title: 'Food';
-
-  public desserts: Dessert[] = [
-    { name: 'Frozen yogurt', calories: 159, fat: 6, carbs: 24, protein: 4 },
-    { name: 'Ice cream sandwich', calories: 237, fat: 9, carbs: 37, protein: 4 },
-    { name: 'Eclair', calories: 262, fat: 16, carbs: 24, protein: 6 },
-    { name: 'Cupcake', calories: 305, fat: 4, carbs: 67, protein: 4 },
-    { name: 'Gingerbread', calories: 356, fat: 16, carbs: 49, protein: 4 },
-    { name: 'Pizza Crunch', calories: 1000, fat: 1000, carbs: 1000, protein: 0 },
-    { name: 'Cheese', calories: 159, fat: 6, carbs: 24, protein: 4 },
-    { name: 'Cake', calories: 262, fat: 16, carbs: 24, protein: 6 },
-    { name: 'Strudel', calories: 305, fat: 4, carbs: 67, protein: 4 },
-    { name: 'Brownie', calories: 356, fat: 16, carbs: 49, protein: 4 },
-    { name: 'Frozen yogurt', calories: 159, fat: 6, carbs: 24, protein: 4 },
-    { name: 'Ice cream sandwich', calories: 237, fat: 9, carbs: 37, protein: 4 },
-    { name: 'Eclair', calories: 262, fat: 16, carbs: 24, protein: 6 },
-    { name: 'Cupcake', calories: 305, fat: 4, carbs: 67, protein: 4 },
-    { name: 'Gingerbread', calories: 356, fat: 16, carbs: 49, protein: 4 },
-    { name: 'Frozen yogurt', calories: 159, fat: 6, carbs: 24, protein: 4 },
-    { name: 'Ice cream sandwich', calories: 237, fat: 9, carbs: 37, protein: 4 },
-    { name: 'Eclair', calories: 262, fat: 16, carbs: 24, protein: 6 },
-    { name: 'Cupcake', calories: 305, fat: 4, carbs: 67, protein: 4 },
-    { name: 'Gingerbread', calories: 356, fat: 16, carbs: 49, protein: 4 },
-  ];
-
-  public displayedColumns = ['name', 'calories', 'fat', 'carbs', 'protein'];
+  public countriesDictionary: Dictionary;
+  public regionDictionary: Dictionary;
+  public micronutrientsDictionary: Dictionary;
+  public popGroupOptions = new Array<PopulationGroup>();
+  public chartData;
+  public displayedColumns = ['name', 'value'];
   public dataSource = new MatTableDataSource();
 
-  constructor() { }
+  constructor(private currentDataService: CurrentDataService, private quickMapsService: QuickMapsService) {}
 
   ngOnInit(): void {
-    this.dataSource = new MatTableDataSource(this.desserts);
+    void this.currentDataService
+      .getTopFood(
+        this.quickMapsService.countryDict,
+        this.quickMapsService.micronutrientDict,
```

Comment:
Here @jon571

Reply:
github didn't find it as a reference :-(